HelpSource/Tutorials/QT/things_you_wanna_know.schelp

```diff
@@ -0,0 +1,61 @@
+summary:: Miscellaneous things you probably want to know about QT.
+categories:: Tutorials>QT
+
+section:: Rendering isn't always available
+
+SuperCollider provides some useful classes to automate processes. Some of them are associated to a link::Classes/Clock:: : mainly link::Classes/Routine:: and link::Classes/Task::.
+
+strong::QT rendering system is unavailable within the context of clock associated threads::. In other words, you cannot 'directly' modify a widget within a link::Classes/Routine:: or a link::Classes/Task::.
+
+Instead, you need to defer the modification by strong::encapsulating it in a deferred function::, that will be executed once SuperCollider resume its main context. Doing otherwise will crash the program :
+
+code::
+(
+var win = Window();
+var slider= Slider();
+var value = 0;
+
+var routine = Routine({
+	loop {
+		value = 101.rand / 100;
+		// Interacting with a widget from within a routine :
+		// instructions must be nested in a function, and deferred
+		{ slider.value_(value); }.defer;
+		0.5.wait;
+	};
+}).play;
+
+win.layout_(
+	VLayout()
+	.add(slider)
+);
+
+CmdPeriod.doOnce({
+	routine.stop;
+	win.close;
+});
+
+win.front;
+)
+::
+
+section:: QT limitations
+
+This remark mostly concerns debugging and issues understanding. Since strong::QT is a third-party program integrated within SuperCollider::, some of its limitations are unrelated to SuperCollider itself.
+
+As an example, QT5 link::Classes/QImage:: has a strict pixel limit and cannot exceed 32767x32767 pixels. This 'issue' being only related to QT, it is documented on QT sources/forums, but not on SuperCollider sources/forums. In other words, working with QT within the context of SuperCollider might also include using QTs documentation directly.
+
+section:: Pen usage and resource management
+
+The link::Classes/Pen:: class allows the user to define a custom code::drawFunc:: to a link::Classes/View::.
+
+link::Classes/Pen:: usage is a strong::two-steps process:: : the user strong::first 'describes' the shape:: he wants to draw (using methods such as code::fillRect:: or code::lineTo::), strong::then renders the shape::, using either code::fill::, code::stroke::, or code::fillStroke::.
+
+Depending on the hardware, strong::when drawing very complex shapes:: (for a example a million lines), strong::some issues might arise:: :
+
+Drawing instructions are cached until they are drawn. This means that calling code::lineTo:: a million times before calling code::stroke:: might saturate the computers RAM before the code::stroke:: instruction effectively takes place and frees the cached instructions.
+
+On the other hand, calling successively code::lineTo:: then code::stroke:: a million time might be time inefficient as several code::lineTo:: could have been grouped together before being drawn on screen.
+
+The best solution in this case would be to test several configurations to balance RAM/CPU usage in the most efficient way.
```

Comment:
Maybe link to the discussions where this came up for more detail, e.g. https://github.com/supercollider/supercollider/issues/3449

Reply:
I like the idea of documentation being autonomous, requiring the least references possible. And I think references should point to other documentation texts, and this one is an issue.
Tbh, I don't consider #3449 being a real issue, and I think it could've been closed way earlier. But it still is open, and there's so many issues currently pending that every one we can close might be considered a (small) victory. I can't do much but this I can. That's the only reason why I wrote this paragraph down...

That's my point of view: I don't think linking to this issue is actually helpful in term of knowledge, but I recognize my incompetence about this question. If you think linking this is a good idea, I'll do it.